### DISCUSSION TITLE
PO file not working for controller and annotation

### DISCUSSION THREAD

**Discussion by yassinsayed** [OrchardCMS/OrchardCore#12055] [Q&A] [answered]
I have a custom module in OrchardCore 1.4. The localization PO file works fine with views but is not affected by controller _notifier message or annotation errors; please see the image and advise if I am missing something
![OF file](https://user-images.githubusercontent.com/34830420/180193385-c39c14c0-d8b2-4b7d-8736-50f0363264b5.gif)
 here is the custom module startup
![image](https://user-images.githubusercontent.com/34830420/180193798-e7e70a1f-2418-4c36-9b1f-f9e026191864.png)

**Comment by hishamco** [comment]
Can you share your controller, as well as the class and view that using the annotation attribute? BWT there are many places in the source code where we did a localization in both controllers and data annotations

**Comment by yassinsayed** [comment]
please see the MemberController, and it is the view (index) and model your support is highly appreciated. 
[MemberController.txt](https://github.com/OrchardCMS/OrchardCore/files/9158917/MemberController.txt)
[ar.po.txt](https://github.com/OrchardCMS/OrchardCore/files/9158961/ar.po.txt)

[index.txt](https://github.com/OrchardCMS/OrchardCore/files/9158918/index.txt)
[AlmaModel.txt](https://github.com/OrchardCMS/OrchardCore/files/9158934/AlmaModel.txt)

**Reply by hishamco** [reply]
When I looked into the `ar.po` there's no localization for the controllers, so it will not be localized

**Reply by yassinsayed** [reply]
it is available and existed at the end of the PO file please search for MemberController

#. "MemberController"

msgctxt "OrchardCore.Alma.Controllers.MemberController"
msgid "Your Profile has be updated successfully"
msgstr "تم تحديث البيانات بنجاح"

**Reply by hishamco** [reply]
I just saw it now, it's my bad when I searched on Notepad ;) just another question are using the default localization module and enable it or you have any sort of customization?

**Reply by yassinsayed** [reply]
just the default

**Reply by hishamco** [reply]
Do you have a minimal repo, so I can easily track the issue. Meanwhile I will look into your files

**Reply by hishamco** [reply]
I forgot one point, be sure that the localization file is not overriden by OC, because there's build task the copy the PO localization into Localization folder

**Reply by yassinsayed** [reply]
it weird .... I just enable the localization features and manually create the ar.po file and write the localization lines, no additional customization or code regarding the overridden I will glade if you explain how to confirm it is not.

**Reply by hishamco** [reply]
Could you please share a minimal repo, pointing to the localization issues? Hope to look into it and fix soon

**Reply by yassinsayed** [reply]
[OrchardCore.Alma.zip](https://github.com/OrchardCMS/OrchardCore/files/9160229/OrchardCore.Alma.zip)

here youare the module, after installing the module you need to browse /orchardcore.Alma/member/index to the see the localization only work for view not for controller, the Po file is available

**Reply by yassinsayed** [reply]
realy need your support

**Reply by hishamco** [reply]
I expect you create a GitHub repo to make my changes live if it's required. If you can please create a quick repo contains the above code, otherwise I will have a look to the attached file

**Reply by yassinsayed** [reply]
the repo has been created in the below URL, your support is highly required.

https://github.com/yassinsayed/AlmaCoreCopy2

**Reply by hishamco** [reply]
Thanks, I will have a look after I update the PR that I'm working on

**Comment by hishamco** [comment, marked as answer]
Seems you did unexpected issue, the key is `Your Profile has be updated successfully` not `Your profile has be updated successfully`, the key is case sensitive ;)

**Reply by hishamco** [reply]
Seems you did unexpected issue, the key is `Your Profile has be updated successfully` not `Your profile has be updated successfully`, the key is case sensitive ;)

**Reply by yassinsayed** [reply]
IT working fine now, Thank you for the fast and excellent response, and I apologize for the silly mistake

**Reply by hishamco** [reply]
It was silly but it happen usually :)
